libminifi/include/utils/ThreadPool.h

```diff
@@ -344,11 +364,13 @@ class ThreadPool {
 // notification for new delayed tasks that's before the current ones
   std::condition_variable delayed_task_available_;
 // map to identify if a task should be
-  std::map<TaskId, bool> task_status_;
+  std::map<TaskId, TaskState> task_status_;
 // manager mutex
   std::recursive_mutex manager_mutex_;
   // thread pool name
   std::string name_;
+  // map of paused tasks moved from worker queue that should not run at the time
+  std::map<TaskId, Worker<T>> paused_tasks_;
```

Comment:
there could be multiple tasks with the same `TaskId` when the maxConcurrentTasks is greater than 1, this way we could discard tasks on a pause + resume

Reply:
Thanks for the info, I was not aware of that. I modified the map to be able to hold multiple tasks for a taskid in [cd5817c](https://github.com/apache/nifi-minifi-cpp/pull/974/commits/cd5817cd7254305b382734d954ebc5b659453519)